model/src/w3gdatmd.F90

```diff
@@ -1566,88 +1501,59 @@ SUBROUTINE W3NMOD ( NUMBER, NDSE, NDST, NAUX )
     !/ End of W3NMOD ----------------------------------------------------- /
     !/
   END SUBROUTINE W3NMOD
-  !/ ------------------------------------------------------------------- /
+
+  !> Initialize an individual spatial grid at the proper dimensions.
+  !>
+  !> Allocate directly into the structure array GRIDS. Note that
+  !> this cannot be done through the pointer alias!
+  !>
+  !> @note
+  !> - Grid dimensions apre passed through parameter list and then
+  !>   locally stored to assure consistency between allocation and
+  !>   data in structure.
+  !> - W3SETG needs to be called after allocation to point to
+  !>   proper allocated arrays.
+  !>
+  !> Switches: !/S    Enable subroutine tracing.
+  !>
+  !> ## Subroutine History
+  !> Date | Modification | Version
+  !> -----|--------------|--------
+  !> 24-Jun-2005 | Origination.                        | 3.07
+  !> 18-Jul-2006 | Add input grids.                    | 3.10
+  !> 05-Oct-2006 | Add filter to array pointers.       | 3.10
+  !> 02-Feb-2007 | Add FLAGST.                         | 3.10
+  !> 30-Oct-2009 | Implement run-time grid selection. (W. E. Rogers & T. J. Campbell, NRL) | 3.14
+  !> 30-Oct-2009 | Implement curvilinear grid type. (W. E. Rogers & T. J. Campbell, NRL)    | 3.14
+  !> 30-Oct-2009 | Implement unstructured grids        | 3.14.1
+  !> 03-Sep-2012 | Clean up of UG grids                | 4.08
+  !> 10-Dec-2014 | Add checks for allocate status      | 5.04
+  !>
+  !> @param[in] imod Model number to point to.
+  !> @param[in] mx Like NX in data structure.
+  !> @param[in] my Like NY in data structure.
+  !> @param[in] msea Like NSEA in data structure.
+  !> @param[in] ndse Error output unit number.
+  !> @param[in] ndst Test output unit number.
+#ifdef W3_SMC
+  !> @param[in] mcel ???
```

Comment:
@MatthewMasarik-NOAA do you know what these parameters are? They are not documented in the source code.

Reply:
@ukmo-ccbunney  would be a good POC for questions on SMC

Reply:
@ukmo-ccbunney, we were curious if you happen to know descriptions for these SMC grid parameters?

Reply:
@edwardhartnett - here are the descriptions for the SMC specific variables:

MCel - number of SMC cells
MUFc - number of SMC U-face connections
MVFc - number of SMC V-face connections
MRLv - number of SMC levels
MBSMC - number of SMC boundary points 
MARC - number of SMC Arctic cells
MBAC - number of SMC Arctic boundary cells
MSPEC - Number of discrete spectral bins (not SMC specific)